pkg/app/api/api/web_api.go

```diff
@@ -313,6 +319,9 @@ func (a *WebAPI) getPiped(ctx context.Context, pipedID string) (*model.Piped, er
 		a.logger.Error("failed to get piped", zap.Error(err))
 		return nil, status.Error(codes.Internal, "Failed to get piped")
 	}
+	if piped.ProjectId != projectID {
+		return nil, status.Error(codes.PermissionDenied, "requested piped doesn't belong to the project you logged in")
```

Comment:
nit: "requested" -> "Requested"
(The web client is using this message to show to our users.)